estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,70 @@
+from datetime import timedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = "Property offer"
+
+    _sql_constraints = [
+        ('check_positive_offer_price', 'CHECK(price > 0)', "The offer price must be strictly positive."),
+    ]
+    _order = "price desc"
+
+    price = fields.Float(string="Price")
+    status = fields.Selection(
+        string="Status",
+        selection=[('accepted', "Accepted"), ('refused', "Refused")],
+        copy=False,
+    )
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', string="Property", required=True)
+    validity = fields.Integer(string="Validiy (days)", default=7)
+    date_deadline = fields.Date(
+        string="Deadline Date",
+        compute='_compute_date_deadline',
+        inverse='_inverse_date_deadline',
+    )
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type", related='property_id.property_type_id')
+
+    @api.depends('create_date', 'validity')
+    def _compute_date_deadline(self):
+        for offer in self:
+            create_date = offer.create_date or fields.Datetime.now()
+            offer.date_deadline = create_date.date() + timedelta(days=offer.validity)
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            create_date = offer.create_date or fields.Datetime.now()
+            offer.validity = (offer.date_deadline - create_date.date()).days
+
+    def action_accept_offer(self):
+        for offer in self:
+            if offer.status == 'accepted':
+                raise UserError(_("You already accepted the offer."))
+            if offer.property_id.buyer_id:
+                raise UserError(_("Only one offer can be accepted."))
+            offer.status = 'accepted'
+            offer.property_id.selling_price = offer.price
+            offer.property_id.buyer_id = offer.partner_id
+            for offer in offer.property_id.offer_ids:
```

Comment:
This could cause some issues
```suggestion
        for offer in self:
            ...
            for property_offer in offer.property_id.offer_ids:
                if offer != property_offer:
                    ...
```